src/activity_item_handling.cpp

```diff
@@ -2941,20 +2940,19 @@ static cata::optional<tripoint> find_refuel_spot_trap( const std::vector<tripoin
     return {};
 }
 
-void find_auto_consume( player &p, const bool food )
+bool find_auto_consume( player &p, const bool food )
 {
+    // returen false if there is no point searching again while the activity is still happening.
```

Comment:
Even if this function returns false, there's still the possibility of the situation changing so that the player can consume an item again. For example, the item may defrost due to a change in temperature, the player may become empty-stomached after being thirsty but engorged, etc. I'd suggest adding a timer to the activity to reset the `no_food_nearby_for_auto_consume` flags once, say, 30 minutes.